src/App.tsx

```diff
@@ -15,48 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum Sortype {
+  Alphabetic = 'alphabetic',
+  Length = 'length',
+  Empty = '',
+}
+
+function getPreaparedGoods(goods: string[],
+  sortType: Sortype,
+  reversed: boolean) {
+  const preparedGoods: string[] = [...goods];
+
+  preparedGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case Sortype.Alphabetic:
+        return good1.localeCompare(good2);
+
+      case Sortype.Length:
+        return good1.length - good2.length;
+
+      default: return 0;
+    }
```

Comment:
you have only 2 conditions, there is no necessary use `switch { case }` (not critical)